extensions/widgets/segment_new/segment_new.lcb

```diff
@@ -0,0 +1,716 @@
+			repeat with tX from 1 up to mNumSegments
+			
+				if the click position is within fetchBounds(tX) then
+					
+					put the empty list into mSelectedSegments
+					push tX onto back of mSelectedSegments
+					
+					put element tX of mSegmentNames into tLabel
+					dispatch command "segmentSelected" to my script object with [tLabel]
+				end if
+				
+			end repeat	
+		end if
+		
+	else
+		-- if can multiselect, then select the clicked-on segment if it is unselected or unselect if its selected
+		-- cannot deselect a segment if it is the only selected segment
+		
+		repeat with tX from 1 up to mNumSegments
+			if the click position is within fetchBounds(tX) then
+		
+				variable tIsIn
+				put tX is in mSelectedSegments into tIsIn
+		
+				if tIsIn then
+					if (the number of elements in mSelectedSegments) > 1 then
+				
+						put 1 into tCount
+						variable tElement
+						repeat for each element tElement in mSelectedSegments
+							if tElement = tX then
+								exit repeat
+							end if
+							add 1 to tCount
+						end repeat
+					
+						delete element tCount of mSelectedSegments
+						put element tX of mSegmentNames into tLabel
+						dispatch command "segmentUnselected" to my script object with [tLabel]
+						
+					end if
+				else
+					put element tX of mSegmentNames into tLabel
+					dispatch command "segmentSelected" to my script object with [tLabel]
+					push tX onto back of mSelectedSegments
+						
+				end if
+			end if
+		end repeat
+	end if
+	
+	redraw all
+end handler
+----------
+
+----------
+-- this handler draws the segments
+private handler drawSegments() as undefined
+	
+	variable tX as integer
+	variable tLabel as string
+	variable tWidth as real
+	variable tIsIn as boolean
+	
+	put 0 into mLeft
+	set the font of this canvas to fetchFont()
+	
+	repeat with tX from 1 up to mNumSegments
+		put fetchWidth(tX) into tWidth
+		put tX is in mSelectedSegments into tIsIn
+		
+		if tIsIn then
+	
+			-- fill in the background of the selected segment(s)
+			set the paint of this canvas to fetchPaint("lines")
+			if tX = 1 then
+				-- rounded rectangle segment
+				fill rounded rectangle path of rectangle [mLeft, 0, mLeft+tWidth, my height] with radius mRadius on this canvas
+				fill rectangle path of rectangle [mLeft+tWidth, 0, mLeft+tWidth*0.8, my height] on this canvas
+				
+			else if tX = mNumSegments then
+				-- rounded rectangle segment
+				fill rounded rectangle path of rectangle [mLeft, 0, mLeft+tWidth, my height] with radius mRadius on this canvas
+				fill rectangle path of rectangle [mLeft, 0, mLeft+tWidth*0.25, my height] on this canvas
+				
+			else
+				fill rectangle path of rectangle [mLeft, 0, mLeft+tWidth, my height] on this canvas
+			end if
+			--
+			
+			-- draw the label(s) in the selected segment(s)
+			if mSegmentDisplay is "icon" then
+				execute script ("return numtocodepoint(" & element tX of mSelectedIcons & ")")
```

Comment:
'parsed as number' only works with base 10 strings at the moment (its strict) so you need to base convert first...

If you look in one of the other widgets which I updated you can crib code from there.